batchflow/notifier.py

```diff
@@ -0,0 +1,315 @@
+            - `a` stands for automatic choise of appropriate tqdm bar.
+            - `j` stands for graph drawing as a progress bar.
+            - `t` or True for standard text tqdm is used.
+            - otherwise, no progress bar will be displayed. Note that iterations,
+            as well as everything else (monitors, variables, logs) are still tracked.
+    total, batch_size, n_iters, n_epochs, length : int
+        Parameters to calculate total amount of iterations.
+    drop_last : bool
+        Whether the last batch of data is dropped from iterations.
+    variables : str, :class:`.NamedExpression` or sequence of them
+        Allows to set trackable entities from the pipeline the Notifier is used in:
+        If str, then stands for name of the variable to get from the pipeline.
+        If any of the named expressions, then evaluated with the pipeline.
+    monitors : str, :class:`.Monitor` or sequence of them
+        Allows to monitor resources. Strings should be registered aliases for monitors like `cpu`, `gpu`, etc.
+    monitor_kwargs : dict
+        Parameters of monitor creation like `frequency`, `pid`, etc.
+    plot : bool
+        If True, then tracked data (usually list of values like memory usage or loss over training process)
+        is dynamically tracked on graphs. Note that rendering takes a lot of time.
+    window : int
+        Allows to plot only the last `window` values from every tracked container.
+    layout : str
+        If `h`, then subplots are drawn horizontally; vertically otherwise.
+    figsize : tuple of numbers
+        Total size of drawn figure.
+    *args, **kwargs
+        Positional and keyword arguments that are used to create underlying progress bar.
+    """
+    def __init__(self, bar=None, *args,
+                 total=None, batch_size=None, n_iters=None, n_epochs=None, drop_last=False, length=None,
+                 variables=None, monitors=None, monitor_kwargs=None, file=None,
+                 plot=False, window=None, layout='h', figsize=None, **kwargs):
+        data_generators = []
+
+        # Prepare variables
+        if variables is not None:
+            variables = variables if isinstance(variables, (tuple, list)) else [variables]
+            data_generators.extend(variables)
+
+        # Prepare monitors
+        if monitors is not None:
+            monitors = 'MemoryMonitor' if monitors is True else monitors
+            monitors = [monitors] if not isinstance(monitors, (tuple, list)) else monitors
+            monitors = [MONITOR_ALIASES[name.lower()](**(monitor_kwargs or {})) if isinstance(name, str) else name
+                        for name in monitors]
+            data_generators.extend(monitors)
+            self.has_monitors = True
+        else:
+            self.has_monitors = False
+
+        # Prepare file log
+        self.file = file
+        if self.file:
+            with open(self.file, 'w') as _:
+                pass
```

Comment:
You're just opening and closing the file, right? What for? Am I missing something?

Reply:
To clear the file contents (note the `w` mode)